plugins/README.md

```diff
@@ -60,6 +60,9 @@ for your own plugin. Extend them to fit your particular use case.
     expressions, compiling them at plugin initialization.
 *   [Overwrite HTTP request & response headers](samples/overwrite_header):
     Overwrite a header on both the client request and server response paths.
+*   [Normalize a HTTP header on request](samples/normalize_header): Check the host
```

Comment:
Device classification isn't typically done using the "host" header (which just reflects the authority portion of the URL). Instead in the past it was done based on the "User-agent" request header, which is starting to be replaced by HTTP Client Hints: [https://developer.chrome.com/docs/privacy-security/user-agent-client-hints](https://www.google.com/url?q=https://developer.chrome.com/docs/privacy-security/user-agent-client-hints&sa=D&source=docs&ust=1723153515044619&usg=AOvVaw2UuOZfnAjN_eUX6sfCP9I9)

I'd propose that instead we have the plugin implement onRequestHeaders to do the following:

* if the "Sec-CH-UA-Mobile" request header is present and has value "?1", then set "client-device-type" header to "mobile".
* Otherwise, if the "User-Agent" request header is present and contains the substring "mobile" (case insensitive), then set "client-device-type" header to "mobile"
* Otherwise, set "client-device-type" request header to "unknown".

Reply:
Done.